R/utils.R

```diff
@@ -0,0 +1,335 @@
+
+  # Apply the OS-conditional change
+  if (.Platform$OS.type == 'windows') {
+    qmd_pattern <- stringr::str_replace_all(qmd_pattern, '\n', '\r\n')
+    qmd_replacement <- stringr::str_replace_all(qmd_replacement, '\n', '\r\n')
+    
+    rmd_pattern <- stringr::str_replace_all(rmd_pattern, '\n', '\r\n')
+    rmd_replacement <- stringr::str_replace_all(rmd_replacement, '\n', '\r\n')
+  }
+  #-----------------------------------------------------------------------------------
+
+  
+  # Append Vignette builder to DESCRIPTION file & modify YAML content
+  if (is_quarto_project) { # Quarto project
+    
+    # Replace the YAML pattern with the new structure for Quarto vignette:
+    readr::write_file(
+      x = stringr::str_replace(
+        string = readr::read_file("vignettes/analysis.qmd"),
+        pattern = qmd_pattern,
+        replacement = qmd_replacement
+      ), 
+      file = "vignettes/analysis.qmd"
+    )
+
+  } else { # Rmd project
+    
+    # Replace the YAML pattern with the new structure for Rmd vignette:
+    readr::write_file(
+      x = stringr::str_replace(
+        string = readr::read_file("vignettes/analysis.Rmd"),
+        pattern = rmd_pattern,
+        replacement = rmd_replacement
+        ), 
+      file = "vignettes/analysis.Rmd"
+    )
+  }
+}
+
+
+#----------------------------------------------------------------------------------------
+#' Build and convert to package
+#' @description
+#' This helper functon will convert an ordinary rUM project into a rUM package. It 
+#' modifies the vignettes/analysis.* YAML header with content appropriate to build 
+#' the respective vignette using the correct engine.
+#' 
+#' This was the previously-named internal function \code{.run_me_first()}.
+#' 
+#' @param path Character. Path where to store the document template, either project root
+#' or vignettes folder for package projects.
+#' @param is_quarto_project Logical. Determines Quarto versus RMarkdown project.
+#' 
+#' @noRd
+.rUM_package_builder <- function(path, is_quarto_project) {
+
+  .add_notes_to_Rbuildignore(path = path)
+
+  # Capture current directory and return to it at the end of this function
+  current_wd <- getwd()
+  # Move to new project location
+  setwd(path)
+  
+  .rUM_build_description(is_quarto_project = is_quarto_project)
+
+  # Create vignettes/.gitignore & write "*.html" & "*.R"
+  writeLines("*.html\n*.R", con = "vignettes/.gitignore")
+
+  # Add inst folder & .gitkeep file
+  dir.create("inst", recursive = TRUE)
+  writeLines("# Do not edit by hand", con = "inst/.gitkeep")
+  # Needed to prevent devtools::check() warning:
+  cat("inst/.gitkeep\n", file = ".Rbuildignore", append = TRUE)
+
+  # Alter document template YAML for vignette builder
+  .rUM_modify_for_vignette(is_quarto_project = is_quarto_project)
+
+  # Return to original location where rUM::make_project() was executed
+  setwd(current_wd)
```

Comment:
Add ending comment here related to wd above

Reply:
same